pkg/cvo/sync_worker.go

```diff
@@ -818,6 +839,10 @@ func (w *SyncWorker) apply(ctx context.Context, work *SyncWork, maxWorkers int,
 					klog.V(2).Infof("Skipping precreation of %s as unmanaged", task)
 					continue
 				}
+				if !capability.ResourceEnabled(resourceCapabilities[i], work.Capabilities.EnabledCapabilities) {
+					klog.V(2).Infof("Skipping precreation of %s, part of disabled capability", task)
```

Comment:
I'd rather be more explicit and have `capability.ResourceEnabled` return an `error` when we decide to skip, explaining why.  Then we can `"Skipping precreation of %s: %s", task, err)` here.

Reply:
Why have it differ from the other checks in this same context?  `ResourceEnabled` is simply that, is this resource enabled given it's configured capabilities, with no knowledge of the impact of it being enabled or not. If we really want a method returning an error I would create another but to me it's an invalid logic shift from `SyncWorker.apply` to `capability` package.

Reply:
Okay, so I see now. Have it list exactly which capability(s) is preventing enablement?